central/image/service/service_impl.go

```diff
@@ -296,6 +301,45 @@ func (s *serviceImpl) ScanImageInternal(ctx context.Context, request *v1.ScanIma
 	return internalScanRespFromImage(img), nil
 }
 
+// updateImageFromRequest will update the name of existing image with the one from the request
+// if the names differ and the metadata for the existing image was unable to be pulled previously.
+// Returns true if an update was made, false otherwise.
+func updateImageFromRequest(existingImg *storage.Image, reqImgName *storage.ImageName) bool {
+	if !features.UnqualifiedSearchRegistries.Enabled() || reqImgName == nil {
+		// The need for this behavior is associated with the use of unqualified search
+		// registries or short name aliases (currently), if the feature is disabled
+		// do not modify the name.
+		return false
+	}
+
+	if existingImg.GetMetadata() != nil {
+		// If metadata exists, then the existing image name is likely valid, no update needed.
```

Comment:
checking the notes may be another option, but this makes sense. If we were able to get metadata, then it's likely the image is valid (and any kind of scan failure was due to some other reason). Just sanity checking this is `nil` vs potentially empty?

Reply:
Chose to do the metadata `nil` check vs. notes for a few reasons:
- We had an issue recently where the notes and metadata were not in sync which is why I opted to go with checking the actual data vs. the note.
- The enricher also checks for [`metadata == nil`](https://github.com/stackrox/stackrox/blob/eee6f455f94e56b05ed048d9b857a941187cc966/pkg/images/enricher/metadata.go#L47).  
- I expect notes to be changed in the future, for example, to indicate 'no metadata' along with a reason (the actual error message).  So felt long term may be the better option to not rely on the notes as is.

> Just sanity checking this is nil vs potentially empty?

Yes, the enricher will only set the metadata object if the it was pulled successfully, otherwise its left at its default value, which is `nil`.